src/sys/unix/selector/epoll.rs

```diff
@@ -31,7 +31,28 @@ impl Selector {
         #[cfg(not(target_os = "android"))]
         let flag = libc::EPOLL_CLOEXEC;
 
-        syscall!(epoll_create1(flag)).map(|ep| Selector {
+        // Illumos and Redox have epoll_create1() function, but no epoll syscalls.
+        #[cfg(any(target_os = "illumos", target_os = "redox"))]
+        let ep = syscall!(epoll_create1(flag))?;
+
+        // Try epoll_create1 syscall with an epoll_create fallback.
+        #[cfg(not(any(target_os = "illumos", target_os = "redox")))]
+        let ep = syscall!(syscall(libc::SYS_epoll_create1, flag))
```

Comment:
We're not going to use raw system calls. Please add the `epoll_create1` function the libc crate instead.

Reply:
> Is there any problem with using a raw system call?

It makes maintenance harder.

But I guess we can make an exception here, could you extract the system call stuff to a new `epoll_create1` function, that for platforms that support it call `libc::epoll_create1` and for others use the raw system interface?